src/Lean/Server/FileWorker/RequestHandling.lean

```diff
@@ -609,18 +624,76 @@ partial def handleWaitForDiagnostics (p : WaitForDiagnosticsParams)
     return t₁.map fun _ => pure WaitForDiagnostics.mk
 
 builtin_initialize
-  registerLspRequestHandler "textDocument/waitForDiagnostics"   WaitForDiagnosticsParams   WaitForDiagnostics      handleWaitForDiagnostics
-  registerLspRequestHandler "textDocument/completion"           CompletionParams           CompletionList          handleCompletion
-  registerLspRequestHandler "textDocument/hover"                HoverParams                (Option Hover)          handleHover
-  registerLspRequestHandler "textDocument/declaration"          TextDocumentPositionParams (Array LocationLink)    (handleDefinition GoToKind.declaration)
-  registerLspRequestHandler "textDocument/definition"           TextDocumentPositionParams (Array LocationLink)    (handleDefinition GoToKind.definition)
-  registerLspRequestHandler "textDocument/typeDefinition"       TextDocumentPositionParams (Array LocationLink)    (handleDefinition GoToKind.type)
-  registerLspRequestHandler "textDocument/documentHighlight"    DocumentHighlightParams    DocumentHighlightResult handleDocumentHighlight
-  registerLspRequestHandler "textDocument/documentSymbol"       DocumentSymbolParams       DocumentSymbolResult    handleDocumentSymbol
-  registerLspRequestHandler "textDocument/semanticTokens/full"  SemanticTokensParams       SemanticTokens          handleSemanticTokensFull
-  registerLspRequestHandler "textDocument/semanticTokens/range" SemanticTokensRangeParams  SemanticTokens          handleSemanticTokensRange
-  registerLspRequestHandler "textDocument/foldingRange"         FoldingRangeParams         (Array FoldingRange)    handleFoldingRange
-  registerLspRequestHandler "$/lean/plainGoal"                  PlainGoalParams            (Option PlainGoal)      handlePlainGoal
-  registerLspRequestHandler "$/lean/plainTermGoal"              PlainTermGoalParams        (Option PlainTermGoal)  handlePlainTermGoal
```

Comment:
This makes me sad, though a code formatter would have done the same...
Actually, why are these explicit type parameters to begin with? With implicits I'd hope these could become one-liners again

Reply:
I don't want to break this API just so that these handlers fit into one line. FWIW, I don't mind the explicit types.

Reply:
As an alternative, in Alloy, I use [type class instances](https://github.com/tydeu/lean4-alloy/blob/5ae885f01218145ac2aac9bc2dd66b99dbb686d3/Alloy/Util/Server/Methods.lean) to infer types for methods. That allows me to keep the registrations on [one line each](https://github.com/tydeu/lean4-alloy/blob/5ae885f01218145ac2aac9bc2dd66b99dbb686d3/Alloy/C/Server.lean#L13-L20) and even automatically infer types when [a handler is provided as a lambda](https://github.com/tydeu/lean4-alloy/blob/5ae885f01218145ac2aac9bc2dd66b99dbb686d3/Alloy/C/Server/Worker.lean#L114-L120).

Reply:
> I don't want to break this API

Reasonable